pynest/nest/tests/compatibility.py

```diff
@@ -65,3 +66,10 @@ def decorator(test_item):
 if not hasattr(unittest.TestCase, 'assertGreater'):
     unittest.TestCase.assertGreater = lambda self, a, b, **kwargs: \
         unittest.TestCase.assertTrue(self, a > b, **kwargs)
+
+# NumPy 1.4.0-
+if not hasattr(numpy, 'fill_diagonal'):
+    def fill_diagonal(a, b):
+        diagonal = numpy.arange(a.shape[0])
+        a[diagonal, diagonal] = b
```

Comment:
You could write those two lines as follows, to handle matrices of all shapes:

    diagonal = [ numpy.arange( min( numpy.shape( a ) ) ) ] * len( numpy.shape( a ) )
    a[ diagonal ] = b